packages/core/src/utils/slang/vendored/slangHelpers.ts

```diff
@@ -0,0 +1,18 @@
+// from https://github.com/NomicFoundation/hardhat-vscode/blob/development/server/src/parser/slangHelpers.ts
+import { getPlatform } from './operatingSystem';
+
+const SUPPORTED_PLATFORMS = [
+  'darwin-arm64',
+  'darwin-x64',
+  'linux-arm64',
+  'linux-x64',
+  'win32-arm64',
+  'win32-ia32',
+  'win32-x64',
+];
+
+export function isSlangSupported() {
+  const currentPlatform = getPlatform();
+
+  return SUPPORTED_PLATFORMS.includes(currentPlatform);
+}
```

Comment:
I feel there should be a workaround to the supported list. Have you considered using `child_process` to ask for the version of slang using `npx`? It'd be similar too:

```suggestion
import cp from 'child_process';

export function isSlangSupported() {
  try {
    cp.execSync('slang --version')
    return true;
  } catch(err) {
    return false;
  }
}
```

Reply:
Changed to import Slang dynamically and catch the error if it fails.